File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/highriskvenuesupload/HighRiskVenuesUploadConfig.java
package uk.nhs.nhsx.highriskvenuesupload;

import uk.nhs.nhsx.core.aws.s3.BucketName;
import uk.nhs.nhsx.core.aws.s3.Locator;
import uk.nhs.nhsx.core.aws.s3.ObjectKey;

public class HighRiskVenuesUploadConfig {

    public final String cloudFrontDistId;
    public final String cloudFrontInvalidationPattern;
    public final Locator locator;

    public HighRiskVenuesUploadConfig(BucketName s3BucketName,
                                      ObjectKey s3ObjKey,
                                      String cloudFrontDistId,
                                      String cloudFrontInvalidationPattern) {

        this.locator = Locator.of(s3BucketName, s3ObjKey);
        this.cloudFrontDistId = cloudFrontDistId;
        this.cloudFrontInvalidationPattern = cloudFrontInvalidationPattern;
    }

}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/highriskvenuesupload/HighRiskVenuesUploadConfig.kt
package uk.nhs.nhsx.highriskvenuesupload

import uk.nhs.nhsx.core.aws.s3.BucketName
import uk.nhs.nhsx.core.aws.s3.Locator
import uk.nhs.nhsx.core.aws.s3.ObjectKey

data class HighRiskVenuesUploadConfig(
    val locator: Locator,
    val cloudFrontDistId: String,
    val cloudFrontInvalidationPattern: String,
) {
    companion object {
        fun of(
            bucket: BucketName,
            key: ObjectKey,
            cloudFrontDistId: String,
            cloudFrontInvalidationPattern: String,
        ) = HighRiskVenuesUploadConfig(Locator.of(bucket, key), cloudFrontDistId, cloudFrontInvalidationPattern)
    }
}
